### PR TITLE
Removed unused array

### PR DESCRIPTION
**Pull request by dk2k**
Removed unused array

### CODE REVIEW

File: server/src/main/java/org/opensearch/search/aggregations/bucket/BucketsAggregator.java [hunk 1]
```diff
@@ -221,10 +221,6 @@ protected final InternalAggregations[] buildSubAggsForBuckets(long[] bucketOrdsT
         }
         InternalAggregations[] result = new InternalAggregations[bucketOrdsToCollect.length];
         for (int ord = 0; ord < bucketOrdsToCollect.length; ord++) {
-            InternalAggregation[] slice = new InternalAggregation[subAggregators.length];
-            for (int i = 0; i < subAggregators.length; i++) {
-                slice[i] = aggregations[i][ord];
-            }
```
Comment (dblock) [inline server/src/main/java/org/opensearch/search/aggregations/bucket/BucketsAggregator.java:227]: Was this intended to be used below? Is it slow to do `aggregations[index][ord]`?

**Review by dblock — APPROVED** [review]

**Review by dblock — CHANGES_REQUESTED** [review]
I wonder if this is hiding a bug and the intermediate value *was* supposed to be used. Maybe someone who understands agg can take a look? @Bukhtawar?

**Comment by sandeshkr419** [comment]
I looked up the commit which added it: https://github.com/opensearch-project/OpenSearch/commit/2f38aeb5e25bf60897ddf87d54357abd92f14ce8#diff-2d2b27a2755fd2a1d43780adc7cda194f64673c4bc655c5a80c1c167cb5bd1e9

Also, tried finding its reference in code, looks like this got added as a dead code in the above commit. 

>Was this intended to be used below? Is it slow to do `aggregations[index][ord]`?
@dblock I don't think slowing things down would have been the intention.


@msfroh Is there an uncompleted implementation of some optimization that we may be overlooking?

File: server/src/main/java/org/opensearch/search/aggregations/bucket/BucketsAggregator.java (hunk 1 above)
Comment (msfroh) [reply server/src/main/java/org/opensearch/search/aggregations/bucket/BucketsAggregator.java:227]: I checked the diff when this was added. I think it was just a mistake. 

I suspect that the previous logic was something like `result[ord] = slice`.

**Comment by dblock** [comment]
> @dblock I don't think slowing things down would have been the intention.

I meant that below the code does `aggregations[index][ord]`. This could have been the beginning of an optimization for that call.

**Comment by dblock** [comment]
I'm leaving this to @msfroh who understands what this code does to decide to merge/not. Thanks for your help @dk2k!

**Review by msfroh — APPROVED** [review]
This looks correct to me. 

It feels very much like this was left over from some refactoring within the commit that introduced this array. 

Specifically, I'm pretty sure that at one point line 229 (now line 225) was:
```
result[ord] = InternalAggregations.from(Arrays.asList(slice));
```

I suspect that either the original author or their reviewer objected to the number of additional objects/arrays being created, so they suggested wrapping list views over existing 2D arrays (but in column order, rather than row order).

Of course, they still left the code that created the unnecessary array -- they just let it get garbage-collected right away.

**Review by sandeshkr419 — APPROVED** [review]

**Review by dblock — APPROVED** [review]

**Comment by dblock** [comment]
@msfroh So ok to merge as is? Hit merge?

**Comment by sandeshkr419** [comment]
@dblock @msfroh Let's hit merge & close on this one?

**Review by msfroh — APPROVED** [review]

**Pull request merged**